lib/bundle-generation.ts

```diff
@@ -0,0 +1,283 @@
+PREFIX sign: <http://mu.semte.ch/vocabularies/ext/handtekenen/>
+PREFIX dct: <http://purl.org/dc/terms/>
+PREFIX nie: <http://www.semanticdesktop.org/ontologies/2007/01/19/nie#>
+PREFIX prov: <http://www.w3.org/ns/prov#>
+PREFIX schema: <http://schema.org/>
+PREFIX besluitvorming: <https://data.vlaanderen.be/ns/besluitvorming#>
+
+SELECT DISTINCT ?report ?physicalFile
+WHERE {
+  VALUES ?reportId { ${reportIds.map(sparqlEscapeString).join(' ')} }
+  GRAPH ${sparqlEscapeUri(config.graph.kanselarij)} {
+    ?report mu:uuid ?reportId .
+    ?report sign:getekendStukKopie/prov:value/^nie:dataSource ?physicalFile .
+    ?report dct:title ?reportName .
+    ?report besluitvorming:beschrijft ?decisionActivity .
+    ?treatment besluitvorming:heeftBeslissing ?decisionActivity .
+    ?treatment dct:subject ?agendaitem .
+    ?agendaitem dct:type ?agendaitemType .
+  }
+  GRAPH ${sparqlEscapeUri(config.graph.public)} { ?agendaitemType schema:position ?typeOrder }
+} ORDER BY ?typeOrder ?reportName`;
+
+  let result;
+  if (viaJob) {
+    result = await querySudo(queryString);
+  } else {
+    result = await query(queryString);
+  }
+  const bindings = result.results.bindings;
+  if (bindings.length === 0) {
+    return null;
+  } else {
+    return bindings.map((binding) => ({
+      report: binding.report.value,
+      physicalFile: binding.physicalFile.value,
+    }));
+  }
+}
+
+async function attachToMeeting(
+  meetingId: string,
+  fileMeta: FileMeta,
+  viaJob: boolean
+) {
+  const pieceUuid = uuid();
+  const pieceUri = `${config.PIECE_RESOURCE_BASE}${pieceUuid}`;
+
+  const documentContainerUuid = uuid();
+  const documentContainerUri = `${config.DOCUMENT_CONTAINER_RESOURCE_BASE}${documentContainerUuid}`;
+
+  const now = new Date();
+
+  const accessLevel = config.INTERN_OVERHEID
+
+  const insertPieceQueryString = `
+  PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+  PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+  PREFIX dossier: <https://data.vlaanderen.be/ns/dossier#>
+  PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#>
+  PREFIX besluitvorming: <https://data.vlaanderen.be/ns/besluitvorming#>
+  PREFIX dct: <http://purl.org/dc/terms/>
+  PREFIX prov: <http://www.w3.org/ns/prov#>
+
+  INSERT {
+    GRAPH ${sparqlEscapeUri(config.graph.kanselarij)} {
+      ?meeting ext:zittingDocumentversie ${sparqlEscapeUri(pieceUri)} .
+      ${sparqlEscapeUri(pieceUri)} a dossier:Stuk ;
+        mu:uuid ${sparqlEscapeString(pieceUuid)} ;
+        dct:title ?pieceName ;
+        besluitvorming:vertrouwelijkheidsniveau ${sparqlEscapeUri(accessLevel)} ;
+        dct:created ${sparqlEscapeDateTime(now)} ;
+        dct:modified ${sparqlEscapeDateTime(now)} .
+      ${sparqlEscapeUri(documentContainerUri)} a dossier:Serie ;
+        mu:uuid ${sparqlEscapeString(documentContainerUuid)} ;
+        dct:created ${sparqlEscapeDateTime(now)} ;
+        dct:type ${sparqlEscapeUri(config.BESLISSINGSFICHE_TYPE)} ;
```

Comment:
Unsure this document-container needs a type? Doesn't seem to be the case on real data

Reply:
https://kaleidos-test.vlaanderen.be/document/655766E48265E66451D4CDBE

Reply:
to be clarified with the users